Ignore blur events fired when the 'Done' button on iOS keyboard is pressed

jmblog opened the pull request:
With a Japanese keyboard on iOS, characters are entered in duplicate when the "Done" button is pressed before selecting conversion candidates.

It's because the blur and focus events are fired consecutively by pressing the 'Done' button.

The `event.relatedTarget` returns `null` when "Done" is clicked, so I use it for detection.

#### Current (Characters are entered in duplicate)
![ios-done-button-current](https://cloud.githubusercontent.com/assets/86085/20419341/3eeb205e-ad99-11e6-9a39-eddfd4cb955b.gif)

#### Expected
![ios-done-button-fixed](https://cloud.githubusercontent.com/assets/86085/20419343/41cf2f36-ad99-11e6-92ca-5776cc9e5d25.gif)

jfschwarz commented:
Thanks for the PR!

In my tests on Chrome and Firefox, the `relatedTarget` of the blur event is always `null`. Effectively, the blur is never handled on your branch, which causes the following problems:

- onBlur is never triggered
- in the single line input the highlightings are shifted after typing long texts and blurring

jmblog commented:
@jfschwarz oops, I missed the problems which you mentioned. I'll consider a different approach...

jfschwarz commented:
Maybe we could get it working by resetting the selectionStart and selectionEnd on focus instead of on blur? (This would possibly require to additionally track in the component whether the input is focused, so that we can prevent calling `setSelection` in `componentDidUpdate`.)

jmblog commented:
@jfschwarz I'm not sure my change is correct, but it seems to work fine and I didn't come across [this issue](https://github.com/effektif/react-mentions/issues/6). Could you review the PR?

jfschwarz commented:
Sorry, but it seems you didn't address any of the issues I listed above. Since `ev.relatedTarget === null` is pretty much always `true` (at least in my tests), please remove this check. Try following the guidance I gave in https://github.com/effektif/react-mentions/pull/125#issuecomment-261543354 and use an additional `isFocused` state (also as a safeguard against https://github.com/effektif/react-mentions/issues/6)

jmblog commented:
I pushed un incomplete commit. Sorry for bothering you. 🙇 

I've removed the `ev.relatedTarget === null` check and also removed the code for resetting selection.

frontendphil commented:
Closing this because of inactivity.
